core/xml.js

```diff
@@ -631,125 +634,164 @@ Blockly.Xml.domToBlockHeadless_ = function(xmlBlock, workspace) {
-        if (!isNaN(width) && !isNaN(height)) {
-          block.commentModel.size = new Blockly.utils.Size(width, height);
-        }
-
-        if (pinned && block.getCommentIcon && !block.isInFlyout) {
-          setTimeout(function() {
-            block.getCommentIcon().setVisible(true);
-          }, 1);
-        }
+        xmlChildNameMap.comment.push(xmlChild);
         break;
       case 'data':
-        block.data = xmlChild.textContent;
+        xmlChildNameMap.data.push(xmlChild);
         break;
       case 'title':
         // Titles were renamed to field in December 2013.
         // Fall through.
       case 'field':
-        Blockly.Xml.domToField_(block, name, xmlChildElement);
+        xmlChildNameMap.field.push(xmlChild);
         break;
       case 'value':
       case 'statement':
-        input = block.getInput(name);
-        if (!input) {
-          console.warn('Ignoring non-existent input ' + name + ' in block ' +
-                       prototypeName);
-          break;
-        }
-        if (childBlockElement) {
-          blockChild = Blockly.Xml.domToBlockHeadless_(childBlockElement,
-              workspace);
-          if (blockChild.outputConnection) {
-            input.connection.connect(blockChild.outputConnection);
-          } else if (blockChild.previousConnection) {
-            input.connection.connect(blockChild.previousConnection);
-          } else {
-            throw TypeError(
-                'Child block does not have output or previous statement.');
-          }
-        }
-        // Set shadow after so we don't create a shadow we delete immediately.
-        if (childShadowElement) {
-          input.connection.setShadowDom(childShadowElement);
-        }
+        xmlChildNameMap.input.push(xmlChild);
         break;
       case 'next':
-        if (childBlockElement) {
-          if (!block.nextConnection) {
-            throw TypeError('Next statement does not exist.');
-          }
-          // If there is more than one XML 'next' tag.
-          if (block.nextConnection.isConnected()) {
-            throw TypeError('Next statement is already connected.');
-          }
-          blockChild = Blockly.Xml.domToBlockHeadless_(childBlockElement,
-              workspace);
-          if (!blockChild.previousConnection) {
-            throw TypeError('Next block does not have previous statement.');
-          }
-          block.nextConnection.connect(blockChild.previousConnection);
-        }
-        // Set shadow after so we don't create a shadow we delete immediately.
-        if (childShadowElement && block.nextConnection) {
-          block.nextConnection.setShadowDom(childShadowElement);
-        }
+        xmlChildNameMap.next.push(xmlChild);
         break;
       default:
         // Unknown tag; ignore.  Same principle as HTML parsers.
         console.warn('Ignoring unknown tag: ' + xmlChild.nodeName);
     }
   }
 
+  var callInitSvg = false;
```

Comment:
Break this out into helper functions where possible.

Reply:
Done